### PR TITLE
Rework repair menu

### LINKED ISSUE #50048: Streamline the repair interface to show repair/improvement/damage chance during item selection
# Is your feature request related to a problem? Please describe.
When attempting item repairs, the item to be repaired needs to be selected (by hitting Enter) in order to see the repair / improvement chance. From there, you need several key presses to get back to the selection menu if the odds are not in your favor. This is frustrating when trying to find an item to tailor/fabricate on with decent chances of success. More frustrating when there's a pile of raggedy clothes/armor nearby.

# Describe the solution you'd like
Rework the repair interface. In my vision of things, the selection panel you get after (a)ctivating the repair tool:
- MUST display the percentage chance of repair / improvement / failure next to each item.
- MUST use dashes to indicate an item that cannot currently be repaired (no material, etc.). EDIT: using NONE as in the example below is actually much better. So the previous should be read as "MUST indicate that an item currently cannot be repaired".
- SHOULD, if possible, indicate the type of material used in the repair and the quantity  available.
 
I posit that something similar to the (E)at menu should work very well here.

# Describe alternatives you've considered
This will likely be a big interface redesign, and the current one does serve its purpose with reasonable effectiveness.

### PR DESCRIPTION
**Pull request by WelpThatWorked**
#### Summary
Interface "Rework repair menu"

#### Purpose of change
Fix #50048 

#### Describe the solution

Adds extra info to menu to prevent need for repeat menu access.

- Menu displays tool selected, relevant skill, available materials (or NONE), and percent chances for success and damage
- Material listing will show red if insufficient material, success chance will display yellow if success chance is zero, damage chance will display yellow if larger than success chance, and both will be light green if damage chance is zero and success chance is not. 

Fix find_repair_difficulty() making unnecessary object copies. 

#### Describe alternatives you've considered

Some different color schemes, as discussed in the issue. Maybe forcing single column if possible.

#### Additional context

image

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
If you're touching the repair menu, you may also want to add an explicit refit option (instead of the current keep repairing past it being fully repaired).

File: src/game_inventory.cpp [hunk 1]
```diff
@@ -1655,15 +1711,27 @@ class repair_inventory_preset: public inventory_selector_preset
...
+        const player &p;
 };
 
+static std::string get_repair_hint( player &p, const repair_item_actor *actor,
+                                    const item *main_tool )
+{
+    auto hint = std::string();
+    hint.append( string_format( "Tool: <color_cyan>%s</color>", main_tool->display_name() ) );
+    hint.append( string_format( " %s ", LINE_XOXO_S ) );
+    hint.append( string_format( "Skill Used: <color_cyan>%s (%d)</color>",
```
Comment (BrettDong) [inline src/game_inventory.cpp:1723]: These strings need to be translated:
```suggestion
    hint.append( string_format( _( "Tool: <color_cyan>%s</color>" ), main_tool->display_name() ) );
    hint.append( string_format( " %s ", LINE_XOXO_S ) );
    hint.append( string_format( _( "Skill Used: <color_cyan>%s (%d)</color>" ),
```

**Comment by WelpThatWorked** [comment]
> If you're touching the repair menu, you may also want to add an explicit refit option (instead of the current keep repairing past it being fully repaired).

As in, adding a new menu option to the one after the one I've redone, or something else?

**Comment by anothersimulacrum** [comment]
At the moment to refit a piece of clothing, I have to have it fully repaired, and then select 1 2 3 or 4
image
There should be an option to explicitly refit it, regardless of repair status.

**Comment by WelpThatWorked** [comment]
Just a single option to work until it's refit, or multiple, mirroring the existing options?

**Comment by anothersimulacrum** [comment]
I would just do a 'work until refit', but you could also do 'try to refit once' and 'work until refit'. But I am not sure anyone does the 'try once' option.

**Comment by TealcOneill** [comment]
@WelpThatWorked Would you like me to poke the devs about the failed tests on the discord?

**Comment by wapcaplet** [comment]
The test failures appear related to `player` references, which no longer work because the `player` class has been eliminated, with most of its features moved to the `Character` class.

In other words, I think it's mostly a matter of replacing your `player &p` with `Character &p` (or `Character &you`, which is more typical). There are also some minor conflicts with the master branch.

This looks like a cool feature, let me know if you need any help getting it to work!

**Comment by Maleclypse** [comment]
@wapcaplet I've added this to "abandoned but desired PRs" project.

**Comment by Fris0uman** [comment]
Obsoleted by https://github.com/CleverRaven/Cataclysm-DDA/pull/59827

**Comment by stuartsmith78** [comment]
@BrettDong GIven that his has been obsoleted by [#59827](https://github.com/CleverRaven/Cataclysm-DDA/pull/59827), should this be removed from the "Abandoned but Desired Pull Requests" project?

**Pull request closed without merging**